### PR TITLE
UCS/UCT: Updating barrier code

### PR DESCRIPTION
**Pull request by shamisp**
Revisiting aarch64 barriers.
Adding special case for TX2.

Signed-off-by: Pavel Shamis (Pasha) <pasharesearch@gmail.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10372/ for details (Mellanox internal link).

File: src/ucs/arch/aarch64/cpu.h [hunk 1]
```diff
@@ -28,14 +28,35 @@ BEGIN_C_DECLS
...
-#define ucs_memory_bus_fence()        asm volatile ("dsb sy" ::: "memory");
-#define ucs_memory_bus_store_fence()  asm volatile ("dsb st" ::: "memory");
-#define ucs_memory_bus_load_fence()   asm volatile ("dsb ld" ::: "memory");
-#define ucs_memory_bus_wc_flush()
-#define ucs_memory_cpu_fence()        asm volatile ("dmb ish" ::: "memory");
-#define ucs_memory_cpu_store_fence()  asm volatile ("dmb ishst" ::: "memory");
-#define ucs_memory_cpu_load_fence()   asm volatile ("dmb ishld" ::: "memory");
-#define ucs_memory_cpu_wc_fence()     asm volatile ("dmb st" ::: "memory");
+
+#define dmb(op)                       asm volatile ("dmb " #op ::: "memory")
```
Comment (yosefe) [inline src/ucs/arch/aarch64/cpu.h]: ucs_aarch64_dmb(_op)
same for dsb

File: src/ucs/arch/ppc64/cpu.h [hunk 2]
```diff
@@ -27,7 +27,7 @@ BEGIN_C_DECLS
 #define ucs_memory_bus_fence()        asm volatile ("sync"::: "memory")
 #define ucs_memory_bus_store_fence()  ucs_memory_bus_fence()
 #define ucs_memory_bus_load_fence()   ucs_memory_bus_fence()
-#define ucs_memory_bus_wc_flush()
+#define ucs_memory_bus_cacheline_wc_flush()
```
Comment (yosefe) [inline src/ucs/arch/ppc64/cpu.h]: what is the reason to rename?

Comment (shamisp) [reply src/ucs/arch/ppc64/cpu.h]: To indicate that this a special flush that used for pushing out the entire c$ line. You cannot use this flush with any other combination of stores.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10455/ for details (Mellanox internal link).

**Pull request merged**